openedge-plugin/src/main/java/org/sonar/plugins/openedge/foundation/OpenEdgeSettings.java

```diff
@@ -175,6 +189,75 @@ private final void initializeXrefBytes(Settings settings) {
     }
   }
 
+  private final void parseBuildDirectory() {
+    if (settings.getBoolean(Constants.SKIP_RCODE))
+      return;
+
+    AtomicInteger numClasses = new AtomicInteger(0);
+    AtomicInteger numMethods = new AtomicInteger(0);
+    AtomicInteger numProperties = new AtomicInteger(0);
+    // Multi-threaded pool
+    long currTime = System.currentTimeMillis();
+    AtomicInteger numRCode = new AtomicInteger(0);
+    ExecutorService service = Executors.newFixedThreadPool(4);
+    Files.fileTreeTraverser().preOrderTraversal(getBinariesDir()).forEach(f -> {
+      if (f.getName().endsWith(".r")) {
+        numRCode.incrementAndGet();
+        service.submit(() -> {
+          try {
+            RCodeInfo rci = new RCodeInfo(new FileInputStream(f));
+            if (rci.isClass()) {
+              numClasses.incrementAndGet();
+              numMethods.addAndGet(rci.getUnit().getMethods().size());
+              numProperties.addAndGet(rci.getUnit().getProperties().size());
+            }
+          } catch (InvalidRCodeException | IOException caught) {
+            LOG.error("Unable to parse rcode " + f.getAbsolutePath(), caught);
+          }
+        });
+      }
+    });
+
+    // Include PL files in $DLC/gui
+    String dlcInstallDir = settings.getString(Constants.DLC);
+    boolean dlcInPropath = settings.getBoolean(Constants.PROPATH_DLC);
+    if (dlcInPropath && !Strings.isNullOrEmpty(dlcInstallDir)) {
+      currTime = System.currentTimeMillis();
+      File dlc = new File(dlcInstallDir);
+
+      Files.fileTreeTraverser().preOrderTraversal(new File(dlc, "gui")).forEach(f -> {
+        if (f.getName().endsWith(".pl")) {
+          service.submit(() -> {
+            parseLibrary(f);
+          });
+        }
+      });
+    }
+    
+    try {
+      service.shutdown();
+      service.awaitTermination(10, TimeUnit.MINUTES);
+    } catch (InterruptedException caught) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS2142)